File: bindings/go/binding_test.go
package tree_sitter_ocaml_test

import (
	"testing"

	tree_sitter "github.com/tree-sitter/go-tree-sitter"
	tree_sitter_ocaml "github.com/tree-sitter/tree-sitter-ocaml/bindings/go"
)

func TestOCamlGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_ocaml.LanguageOCaml())
	if language == nil {
		t.Errorf("Error loading OCaml grammar")
	}

	sourceCode := []byte("module M = struct let x = 0 end")
	parser := tree_sitter.NewParser()
	defer parser.Close()
	parser.SetLanguage(language)

	tree := parser.Parse(sourceCode, nil)
	if tree == nil || tree.RootNode().HasError() {
		t.Errorf("Error parsing OCaml")
	}
}

func TestOCamlInterfaceGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_ocaml.LanguageOCamlInterface())
	if language == nil {
		t.Errorf("Error loading OCamlInterface grammar")
	}

	sourceCode := []byte("module M : sig val x : int end")
	parser := tree_sitter.NewParser()
	defer parser.Close()
	parser.SetLanguage(language)

	tree := parser.Parse(sourceCode, nil)
	if tree == nil || tree.RootNode().HasError() {
		t.Errorf("Error parsing OCamlInterface")
	}
}

func TestOCamlTypeGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_ocaml.LanguageOCamlType())
	if language == nil {
		t.Errorf("Error loading OCamlType grammar")
	}

	sourceCode := []byte("int list")
	parser := tree_sitter.NewParser()
	defer parser.Close()
	parser.SetLanguage(language)

	tree := parser.Parse(sourceCode, nil)
	if tree == nil || tree.RootNode().HasError() {
		t.Errorf("Error parsing OCamlType")
	}
}

File: bindings/node/binding_test.js
import assert from 'node:assert';
import {describe, it} from 'node:test';
import Parser from 'tree-sitter';

import {ocaml, ocaml_interface, ocaml_type} from './index.js';

describe('OCaml', () => {
  const parser = new Parser();
  parser.setLanguage(ocaml);

  it('should be named ocaml', () => {
    assert.strictEqual(parser.getLanguage().name, 'ocaml');
  });

  it('should parse source code', () => {
    const sourceCode = `
    module M = struct
      let x = 0
    end
    `;
    const tree = parser.parse(sourceCode);
    assert(!tree.rootNode.hasError);
  });
});

describe('OCamlInterface', () => {
  const parser = new Parser();
  parser.setLanguage(ocaml_interface);

  it('should be named ocaml_interface', () => {
    assert.strictEqual(parser.getLanguage().name, 'ocaml_interface');
  });

  it('should parse source code', () => {
    const sourceCode = `
    module M : sig
      val x : int
    end
    `;
    const tree = parser.parse(sourceCode);
    assert(!tree.rootNode.hasError);
  });
});

describe('OCamlType', () => {
  const parser = new Parser();
  parser.setLanguage(ocaml_type);

  it('should be named ocaml_type', () => {
    assert.strictEqual(parser.getLanguage().name, 'ocaml_type');
  });

  it('should parse source code', () => {
    const sourceCode = `
    int list
    `;
    const tree = parser.parse(sourceCode);
    assert(!tree.rootNode.hasError);
  });
});
